### PR TITLE
Remove broken cycle detection algorithm, change 'in' operator to not support date inference

### PR DESCRIPTION
**Pull request by jkodumal**

### CODE REVIEW

File: prereq_cycle_test.go [hunk 1]
```diff
@@ -1,68 +0,0 @@
...
-}
-
-func TestPrereqCycle(t *testing.T) {
-	featureFlagA := newFlagWithPrereq("keyA", "keyB")
-	featureFlagB := newFlagWithPrereq("keyB", "keyC")
-	featureFlagC := newFlagWithPrereq("keyC", "keyA")
-	testFlag(t, "keyA", []FeatureFlag{featureFlagA, featureFlagB, featureFlagC})
-}
-
-func testFlag(t *testing.T, key string, flags []FeatureFlag) {
```
Comment (drichelson) [inline prereq_cycle_test.go]: I think we still want this test.

Comment (jkodumal) [inline prereq_cycle_test.go]: Which test?

Comment (pkaeding) [inline prereq_cycle_test.go]: that's not an actual test, it is the common part DRYed up from the actual (removed) tests.

**Comment by pkaeding** [comment]
lgtm
:+1:
looks good to me
merge away
I approve
canned replies are great.

File: prereq_cycle_test.go (hunk 1 above)
Comment (drichelson) [inline prereq_cycle_test.go]: right- my bad. I saw an assert.

**Comment by drichelson** [comment]
👍

**Pull request merged**